app/registries/foreman/plugin.rb

```diff
@@ -38,21 +38,43 @@ class PluginRequirementError < Foreman::Exception; end
   #   end
   #
   class Plugin
+    DEFAULT_REGISTRIES = {
+      fact_importer: -> { Plugin::FactImporterRegistry.new },
+      report_scanner: -> { Plugin::ReportScannerRegistry.new },
+      report_origin: -> { Plugin::ReportOriginRegistry.new },
+      medium_providers: -> { Plugin::MediumProvidersRegistry.new },
+      graphql_types: -> { Plugin::GraphqlTypesRegistry.new },
```

Comment:
I wondered about this. Does it make sense to store just the class and call `.new` in `initialize_default_registries`. On the one hand, this is more flexible (since in the future you could use parameters to `.new`) but on the other hand, we don't need it now. Have you thought about this?

Reply:
I did and I've ended up with exactly the same thing - this approach being more flexible, but I guess we can go with only the classes for now and change it in the future, given it's only the internal implementation.

Reply:
Used strings - I think sometimes hashes do not like Classes and it should be better to store strings, so I've used class names